packages/std/src/mock.rs

```diff
@@ -492,20 +498,43 @@ impl<C: CustomQuery + DeserializeOwned> MockQuerier<C> {
     }
 }
 
-#[derive(Clone, Default)]
-struct NoWasmQuerier {
-    // FIXME: actually provide a way to call out
+struct WasmQuerier {
+    /// A handler to handle Wasm queries. This is set to a dummy handler that
+    /// always errors by default. Update it via `with_custom_handler`.
+    ///
+    /// Use box to avoid the need of generic type.
+    handler: Box<dyn for<'a> Fn(&'a WasmQuery) -> QuerierResult>,
```

Comment:
This would allow for a closure with some immutable data, but not mutable data, right?

I guess all mutability needs to happen before storing it in the WasmQuerier.